src/node_types.cc

```diff
@@ -51,6 +51,15 @@ static void IsAnyArrayBuffer(const FunctionCallbackInfo<Value>& args) {
     args[0]->IsArrayBuffer() || args[0]->IsSharedArrayBuffer());
 }
 
+static void IsBoxedPrimitive(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Would it be faster to do this in javascript? For example: `typeof arg === 'object' && arg !== null && arg.constructor === String` ?

Reply:
I did not test it but I would not want to use it as it's not a safe check. Someone could easily manipulate the constructor.

```js
const s = new String('test')
s.constructor = null
```

Reply:
Alright, how about:

```js
if (typeof arg === 'object' && arg !== null && Object.prototype.toString.call(arg) === '[object String]')
```

?

Reply:
That is also not a safe check:

```
const a = new String('test')
a[Symbol.toStringTag] = 'Foo'
console.log(Object.prototype.toString.call(a))
// '[object Foo]'
```

I'll move the bigint check in the C++ function down and I'll see if I can work around the performance penalty. Do you expect a lot of people using boxed primitives? Because for all other cases there are now less calls to C++ than before.

Reply:
Seems like the main regression came due to moving the functionality to extra function. It's probably not inlined anymore as other code is hotter and the function is already big. I moved it back in to address the performance regression.